Setup/Program.cs

```diff
@@ -0,0 +1,140 @@
+                Dirs = new[] {installDir},
+                GUID = new Guid("2601159C-2189-433D-9CB1-D3E8A672A9A0"),
+                RegValues = GetRegValues(mainDirPath, InstallDirPath)
+            };
+
+            project.BuildMsi();
+        }
+
+        private static (string, string) GetMainDirs()
+        {
+            var currentDirPath = Environment.CurrentDirectory;
+            var mainDirPath = currentDirPath.Substring(0,
+                currentDirPath.IndexOf(MainDirectoryName, StringComparison.CurrentCulture)
+                + MainDirectoryName.Length + 1);
+            var mainAppDirPath = Path.Combine(mainDirPath, MainAppFilesDirectory);
+            return (mainAppDirPath, mainDirPath);
+        }
+
+        private static WixSharpFile[] GetFiles(string mainDirPath, string mainAppDirPath)
+        {
+            var filePathsSourceFilePath = Path.Combine(mainDirPath, "FilePaths.json");
+
+            using(var filePathsSourceFile =
+                  File.Open(filePathsSourceFilePath, FileMode.Open, FileAccess.Read))
+            {
+                var serializer = new DataContractJsonSerializer(typeof(JsonFilePathsSet));
+                var filePathsSet = (JsonFilePathsSet) serializer.ReadObject(filePathsSourceFile);
+                return filePathsSet.FilePaths
+                    .Select(x => new WixSharpFile(Path.Combine(mainAppDirPath, x)))
+                    .ToArray();
+            }
+        }
+
+        private static RegValue[] GetRegValues(string mainDirPath, string mainAppDirPath)
+        {
+            var mainExeFilePath = Path.Combine(mainAppDirPath, AppName + ".exe");
+
+            RegValue[] res;
+            var registrySetupPath = Path.Combine(mainDirPath, "RegistrySetup.json");
+            using(var registrySetupFile = File.Open(registrySetupPath, FileMode.Open, FileAccess.Read))
+            {
+                var serializer = new DataContractJsonSerializer(typeof(JsonRegKeySet));
+                var regKeySet = (JsonRegKeySet) serializer.ReadObject(registrySetupFile);
+                res = regKeySet.RegistryKey_ValuesSets
+                    .Select(x => x.Values.Select(y => (x.MainKey, x.Path, y.Name, y.Value)))
+                    .SelectMany(x => x)
+                    .Select(x => (x.MainKey, string.Format(x.Path, AppName), x.Name, x.Value))
+                    .Select(x =>
+                    {
+                        var regPath = x.MainKey + '\\' + x.Item2;
+                        string value;
+                        if(PathKeys.ContainsKey(regPath) && PathKeys[regPath].Equals(x.Name))
+                        {
+                            value = string.Format(x.Value, string.Concat("\"", mainExeFilePath, "\""));
+                        }
+                        else
+                        {
+                            value = string.Format(x.Value, AppName);
+                        }
+
+                        return (x.MainKey, x.Item2, x.Name, value);
```

Comment:
To avoid meaningless property names like `Item2`, it is better to use either anon classes `new {Name=value}` or named tuples `(name:value)`

Reply:
Resolved in recent commits